source/common/http/conn_manager_utility.cc

```diff
@@ -122,18 +122,10 @@ void ConnectionManagerUtility::mutateRequestHeaders(
 
   // Generate x-request-id for all edge requests, or if there is none.
   if (config.generateRequestId() && (edge_request || !request_headers.RequestId())) {
-    std::string uuid = "";
-
-    try {
-      uuid = random.uuid();
-    } catch (const EnvoyException&) {
-      // We could not generate uuid, not a big deal.
-      config.stats().named_.failed_generate_uuid_.inc();
-    }
-
-    if (!uuid.empty()) {
-      request_headers.insertRequestId().value(uuid);
-    }
+    // TODO PERF: Write UUID directly to the header map.
+    const std::string uuid = random.uuid();
+    ASSERT(!uuid.empty());
```

Comment:
If it can't happen in real impl, I would just move the ON_CALL and default UUID into the mock itself. This will further reduce boilerplate in the different tests.